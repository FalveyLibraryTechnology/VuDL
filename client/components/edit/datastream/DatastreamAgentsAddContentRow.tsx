import React, { Dispatch, SetStateAction } from "react";
import AddCircleOutline from "@mui/icons-material/AddCircleOutline";
import IconButton from "@mui/material/IconButton";
import DatastreamAgentsContentRow from "./DatastreamAgentsContentRow";
import { useEditorContext } from "../../../context/EditorContext";

export interface Agent {
    role: string;
    type: string;
    name: string;
    notes: Array<string>;
}
export interface DatastreamAgentsAddContentRowProps {
    agent: Agent;
    setAgent: Dispatch<SetStateAction<Agent>>;
    setHasChanges: (hasChanges: boolean) => void;
}

const DatastreamAgentsAddContentRow = ({
    agent,
    setAgent,
    setHasChanges,
}: DatastreamAgentsAddContentRowProps): React.ReactElement => {
    const {
        state: { currentAgents },
        action: { setCurrentAgents },
    } = useEditorContext();
    const onChangeKey = (key: string, value: string | Array<string>) => {
        agent[key] = value;
        setAgent({ ...agent });
        setHasChanges(true);
    };
    const addAgent = () => {
        setCurrentAgents([...currentAgents, agent]);
        setAgent({
            role: "",
            type: "",
            name: "",
            notes: [],
        });
        setHasChanges(true);
    };
    return (
        <DatastreamAgentsContentRow
            agent={agent}
            additionalControls={
                <IconButton disabled={!agent.role || !agent.type || !agent.name} onClick={addAgent}>
                    <AddCircleOutline />
                </IconButton>
            }
            initialExpand={true}
            namesHelperText={
                !agent.role || !agent.type || !agent.name
                    ? "Role, type, and name fields must be filled in order to save the add row."
                    : ""
            }
            onRoleChange={(value) => onChangeKey("role", value)}
            onTypeChange={(value) => onChangeKey("type", value)}
            onNameChange={(value) => onChangeKey("name", value)}
            onNotesChange={(value) => onChangeKey("notes", value)}
        />
    );
};

export default DatastreamAgentsAddContentRow;
