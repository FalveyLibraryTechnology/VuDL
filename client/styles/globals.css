:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
