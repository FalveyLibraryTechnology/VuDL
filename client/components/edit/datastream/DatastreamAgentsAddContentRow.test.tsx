import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DatastreamAgentsAddContentRow from "./DatastreamAgentsAddContentRow";

const mockDatastreamAgentsContentRow = jest.fn();
jest.mock("./DatastreamAgentsContentRow", () => (props) => {
    mockDatastreamAgentsContentRow(props);
    return "DatastreamAgentsContentRow";
});
const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

describe("DatastreamAgentsAddContentRow", () => {
    let agent;
    let props;
    let editorValues;

    beforeEach(() => {
        agent = {
            role: "test1",
            type: "test2",
            name: "test3",
            notes: ["test4"],
        };
        props = {
            agent,
            setAgent: jest.fn(),
            setHasChanges: jest.fn(),
        };
        editorValues = {
            state: { currentAgents: ["test1"] },
            action: { setCurrentAgents: jest.fn() },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("renders", () => {
        const wrapper = shallow(<DatastreamAgentsAddContentRow {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
