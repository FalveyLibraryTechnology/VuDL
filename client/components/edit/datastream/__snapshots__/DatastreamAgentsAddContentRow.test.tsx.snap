// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamAgentsAddContentRow renders 1`] = `
<Component
  additionalControls={
    <ForwardRef(IconButton)
      disabled={false}
      onClick={[Function]}
    >
      <Memo />
    </ForwardRef(IconButton)>
  }
  agent={
    Object {
      "name": "test3",
      "notes": Array [
        "test4",
      ],
      "role": "test1",
      "type": "test2",
    }
  }
  initialExpand={true}
  namesHelperText=""
  onNameChange={[Function]}
  onNotesChange={[Function]}
  onRoleChange={[Function]}
  onTypeChange={[Function]}
/>
`;
